### PR TITLE
#1809 WizardReducer

### LINKED ISSUE #1809: Wizard reducer for managing tabs
## Is your feature request related to a problem? Please describe.

At the moment the wizard component is very simple and just moves tabs. Having a redux reducer would simplify this a lot.

## Describe the solution you'd like

A `WizardReducer` which would have state:

```
{
    isComplete,
    currentTab,
    tabs,    
}
```

with action creators :
- `complete`
- `next`
- `prev`
- `switch`
- `init` (?)

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1809 

## Change summary

- Adds `WizardReducer` and actions to control the wizard component.
- Wanted a `Complete` field.. ended up using that to go a bit off road and disable implement disabling of things after finalising..

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/widgets/Stepper.js [hunk 1]
```diff
@@ -53,8 +53,7 @@ const StepperNumber = ({ step, numberOfSteps, currentStep, title, onPress }) =>
   const lastStep = step === numberOfSteps - 1;
   const completedStep = currentStep > step;
   const Container = completedStep ? TouchableOpacity : View;
-
-  const wrappedOnPress = () => onPress(step - 1);
+  const wrappedOnPress = () => onPress(step);
```
Comment (mark-prins) [inline src/widgets/Stepper.js]: is this a fix? I'm unsure why the `step - 1` was a thing, when the `onPress(step)` looks correct to me

File: src/widgets/Wizard.js [hunk 2]
```diff
@@ -2,31 +2,46 @@
...
+  currentTab: PropTypes.number.isRequired,
+};
+
+const mapStateToProps = state => {
+  const currentTab = selectCurrentTab(state);
+  return { currentTab };
 };
+
+const mapDispatchToProps = dispatch => ({
+  nextTab: tab => dispatch(WizardActions.switchTab(tab)),
```
Comment (mark-prins) [inline src/widgets/Wizard.js]: this is muddling the usage of `nextTab` which in some cases doesn't accept an argument now, and in this usage does. I'd suggest renaming this method / prop or changing it to use `WizardActions.NEXT_TAB`

Comment (joshxg) [reply src/widgets/Wizard.js]: thanks ! 729fb1c904ca57b39110894b85e8439a11cab554

File: src/widgets/Stepper.js (hunk 1 above)
Comment (joshxg) [reply src/widgets/Stepper.js]: uhh.. I can't remember..

File: src/pages/PrescriptionPage.js [hunk 3]
```diff
@@ -13,36 +13,25 @@ import { PrescriberSelect } from '../widgets/Tabs/PrescriberSelect';
...
 
-import {
-  selectItem,
-  selectPrescriber,
-  switchTab,
-  editQuantity,
-} from '../reducers/PrescriptionReducer';
-
-const mapDispatchToProps = dispatch => {
-  const choosePrescriber = prescriberID => dispatch(selectPrescriber(prescriberID));
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/pages/PrescriptionPage.js]: nice! 5467dc9f1547f260d95da632a7c941178bea5c6c

File: src/widgets/PrescriptionCartRow.js [hunk 4]
```diff
@@ -24,20 +24,20 @@ import { UIDatabase } from '../database';
  *
  * Is passed all details needed as this is a simple layout and presentation component.
  *
- * @prop {Object} transactionItem
- * @prop {Func}   onChangeQuantity
- * @prop {Func}   onRemoveItem
- * @prop {Array}  options
- * @prop {Func}   onOptionSelection
+ * @prop {Object} transactionItem    Underlying Transaction Item for this row.s
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/PrescriptionCartRow.js]: 🤣 

2aa048daabcb8d260054da0f83a9ff33793f44e7

File: src/widgets/PrescriptionCart.js [hunk 5]
```diff
@@ -26,12 +26,14 @@ import { removeItem, updateDirection } from '../reducers/PrescriptionReducer';
  * @prop {Func}  onChangeQuantity  Callback when an items quantity is updated.
  * @prop {Func}  onOptionSelection Callback when an option in the dropdown is selected.
  * @prop {Func}  onRemoveItem      Callback when this row is removed.
+ * @prop {Bool}  isDisabled        Indicator if this component should be editable.
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/PrescriptionCart.js]: ec479d46e2cab11b3bb8f1f4b02305e08cecb176

File: src/widgets/DropdownRow.js [hunk 6]
```diff
@@ -30,6 +30,7 @@ import { FlexView } from './FlexView';
  * @prop {Bool}   useSecondaryMenu  Indicator whether the secondary menu should show.
  * @prop {Number} iconSize          The size of the drop down icon.
  * @prop {String} placeholder       Placeholder string value, when no value has been chosen/entered.
+ * @prop {Bool}   isDisabled        Indicator if this component should be editable.
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/DropdownRow.js]: ec479d46e2cab11b3bb8f1f4b02305e08cecb176

File: src/reducers/WizardReducer.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+const initialState = () => ({
+  currentTab: 0,
+  isComplete: false,
+});
+
+export const WizardReducer = (state = initialState(), action) => {
+  const { type } = action;
+
+  switch (type) {
+    case 'Navigation/BACK': {
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/reducers/WizardReducer.js]: yeah, it's the hardware back button being pressed. The actions are from `react-navigation` package :(

**Pull request merged**